e2e/tests/transfer/forwarding_test.go

```diff
@@ -220,3 +221,96 @@ func (s *TransferForwardingTestSuite) TestForwardingWithUnwindSucceeds() {
 		s.AssertPacketRelayed(ctx, chainB, channelBtoC.PortID, channelBtoC.ChannelID, 1)
 	})
 }
+
+func (s *TransferForwardingTestSuite) TestChannelUpgradeForwarding_Succeeds() {
+	ctx := context.TODO()
+	t := s.T()
+	testName := t.Name()
+	t.Parallel()
+
+	relayer := s.CreateDefaultPaths(testName)
+	chains := s.GetAllChains()
+
+	chainA, chainB, chainC := chains[0], chains[1], chains[2]
+
+	opts := s.TransferChannelOptions()
+	opts.Version = transfertypes.V1
+
+	channelAtoB, _ := s.CreatePath(ctx, relayer, chains[0], chains[1], ibc.DefaultClientOpts(), opts, testName)
+	s.Require().Equal(transfertypes.V1, channelAtoB.Version, "the channel version is not ics20-1")
+
+	channelBtoC, _ := s.CreatePath(ctx, relayer, chains[1], chains[2], ibc.DefaultClientOpts(), opts, testName)
```

Comment:
Not a problem with how the test it is right now, but maybe it could be simplified a bit if channel B to C is already `ics20-2` and doesn't need to upgrade.

Reply:
ah that's fine by me! I thought we wanted to have both channels v1 for completeness, I don't mind changing this one :)

Reply:
Whatever you prefer, I don't feel strongly about it. :)